addons/binding/org.openhab.binding.avmfritz/src/main/java/org/openhab/binding/avmfritz/handler/AVMFritzBaseBridgeHandler.java

```diff
@@ -0,0 +1,331 @@
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.OpenClosedType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.avmfritz.internal.ahamodel.AVMFritzBaseModel;
+import org.openhab.binding.avmfritz.internal.ahamodel.DeviceModel;
+import org.openhab.binding.avmfritz.internal.ahamodel.GroupModel;
+import org.openhab.binding.avmfritz.internal.ahamodel.HeatingModel;
+import org.openhab.binding.avmfritz.internal.ahamodel.SwitchModel;
+import org.openhab.binding.avmfritz.internal.config.AVMFritzConfiguration;
+import org.openhab.binding.avmfritz.internal.hardware.FritzahaWebInterface;
+import org.openhab.binding.avmfritz.internal.hardware.callbacks.FritzAhaUpdateXmlCallback;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Abstract handler for a FRITZ! bridge. Handles polling of values from AHA devices.
+ *
+ * @author Robert Bausdorf - Initial contribution
+ * @author Christoph Weitkamp - Added support for AVM FRITZ!DECT 300 and Comet
+ *         DECT
+ * @author Christoph Weitkamp - Added support for groups
+ *
+ */
+public abstract class AVMFritzBaseBridgeHandler extends BaseBridgeHandler implements IFritzHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(AVMFritzBaseBridgeHandler.class);
+
+    /**
+     * Initial delay in s for polling job.
+     */
+    private static final int INITIAL_DELAY = 1;
+    /**
+     * Refresh interval which is used to poll values from the FRITZ!Box web interface (optional, defaults to 15 s)
+     */
+    private long refreshInterval = 15;
+    /**
+     * Interface object for querying the FRITZ!Box web interface
+     */
+    private FritzahaWebInterface connection;
+    /**
+     * Schedule for polling
+     */
+    private ScheduledFuture<?> pollingJob;
+
+    /**
+     * Constructor
+     *
+     * @param bridge Bridge object representing a FRITZ!Box
+     */
+    public AVMFritzBaseBridgeHandler(@NonNull Bridge bridge) {
```

Comment:
See remark on of the other PR's about using `@NonNull`

Reply:
I'll try to add @NonNullByDefault to the handler like suggested in https://github.com/openhab/openhab2-addons/pull/3399#comment-177091695